### DISCUSSION TITLE
Reintroducing Helm Chart for Simplified Kubernetes Deployments

### DISCUSSION THREAD

**Discussion by kersten** [inovex/scrumlr.io#4079] [Proposals]
Hello team,

As someone currently working on deploying Scrumlr.io within a Kubernetes environment, I noticed that a Helm chart had previously been available but was removed as seen in PR [#2808](https://github.com/inovex/scrumlr.io/pull/2808). Understanding that decisions like this are often made with good reason, I am interested in discussing the possibility of reintroducing a Helm chart to facilitate easier Kubernetes deployments.

### Context:
The use of Helm charts is widely recognized for simplifying the management of Kubernetes applications. They provide a structured approach to configuring and managing Kubernetes resources, making it easier for teams to install, upgrade, and maintain their applications consistently.

### Proposal for Discussion:

- **Feasibility:** What were the reasons for the initial removal of the Helm chart, and have these circumstances changed in a way that might support reintroduction?
- **Benefits:** Could we discuss the potential benefits that a Helm chart would bring to current and future users looking to deploy Scrumlr.io in Kubernetes environments?
- **Community Contribution:** If there is interest but limited resources from the core development team for this task, could this be an area where community contributions would be welcomed?

This discussion aims to understand the landscape and potential paths forward to possibly reintroduce this tool, which could significantly streamline Kubernetes deployments for many users.

Kersten

**Comment by Lennart01** [comment]
Hi there,
We originally decided to remove the helm chart because we weren't actively maintaining it and decided it should not be part of the repo.
We had no quality control or tests to verify that it still worked and changed our deployment internally.
As an open source project, we welcome community contributions, so if this is something you're interested in and would be willing to do, we would encourage it.
We will be starting a new docs page (#4085) and want to include some documentation on deployment options (mostly Docker Compose and some kind of Kubernetes deployment (Manifest or Helm).
Important to me would be that it is properly documented and automatically tested so that we can verify that the chart still works as intended. We have been doing some work towards more end-to-end testing and dev deployments, so maybe we can test the chart as part of our automatic dev deployment (e.g. image value).
I do see the benefit of a helm chart as it can be a great way to take a tool for a quick spin and increase adoption.
Internally we mainly rely on CloudnativePG for postres and the official nats chart, so I would suggest that we build the chart on top of these two tools, they have been battle tested in production for over a year now.
Eventually Id also like to see a helm chart repo as part of a github pages deployment for example using [chart-releaser](https://github.com/helm/chart-releaser).
If you need any guidance on how to contribute, feel free to ask 😄.
